src/CodeGeneration.Roslyn.Tool/Program.cs

```diff
@@ -39,9 +40,12 @@ private static async Task<int> Core(string[] args, CancellationToken cancellatio
             IReadOnlyList<string> refs = Array.Empty<string>();
             IReadOnlyList<string> preprocessorSymbols = Array.Empty<string>();
             IReadOnlyList<string> plugins = Array.Empty<string>();
-            string generatedCompileItemFile = null;
-            string outputDirectory = null;
-            string projectDir = null;
+            string? generatedCompileItemFile = null;
+
+            // a value is assigned in the command line declaration (DefineOption), after option parsing there is an explicit null check
+            string outputDirectory = null!;
+            // a value is assigned in the command line declaration (DefineOption), after option parsing there is an explicit null check
+            string projectDir = null!;
```

Comment:
The explicit null check that comes later is good. But it doesn't mean we should define this as nullable up here, where we're deliberately setting it to null. If we set it to null, it should be nullable. Later, after the null check, the compiler will recognize that the value is not null and treat it as a non-null value, which should mean there's no benefit to making this `string` here instead of `string?`